ada_feeding/ada_feeding/watchdog/ft_sensor_condition.py

```diff
@@ -0,0 +1,180 @@
+        self.ft_sensor_topic = self._node.resolve_topic_name("~/ft_topic")
+        self.ft_sensor_subscription = self._node.create_subscription(
+            WrenchStamped,
+            self.ft_sensor_topic,
+            self.__ft_sensor_callback,
+            rclpy.qos.QoSPresetProfiles.SENSOR_DATA.value,
+        )
+
+    def __ft_sensor_callback(self, msg: WrenchStamped) -> None:
+        """
+        Callback function for the force-torque sensor topic.
+
+        Parameters
+        ----------
+        msg: WrenchStamped
+            The message received from the force-torque sensor topic.
+        """
+        # Convert the message to a numpy array
+        ft_data = np.array(
+            [
+                msg.wrench.force.x,
+                msg.wrench.force.y,
+                msg.wrench.force.z,
+                msg.wrench.torque.x,
+                msg.wrench.torque.y,
+                msg.wrench.torque.z,
+            ]
+        )
+        ft_time = Time.from_msg(msg.header.stamp)
+
+        # Get the last unique values and the last unique values timestamp
+        with self.last_unique_values_lock:
+            last_unique_values = self.last_unique_values
+            last_unique_values_timestamp = self.last_unique_values_timestamp
+
+        # If this is the first message received, initialize the last unique values
+        # and the last unique values timestamp
+        if last_unique_values is None:
+            last_unique_values = ft_data
+            last_unique_values_timestamp = np.repeat(ft_time, ft_data.shape[0])
+        else:
+            # Update the last unique values
+            dimensions_that_havent_changed = np.isclose(last_unique_values, ft_data)
+            last_unique_values = np.where(
+                dimensions_that_havent_changed,
+                last_unique_values,
+                ft_data,
+            )
+            last_unique_values_timestamp = np.where(
+                dimensions_that_havent_changed,
+                last_unique_values_timestamp,
+                ft_time,
+            )
+
+        # Set the last unique values and the last unique values timestamp
+        with self.last_unique_values_lock:
+            self.last_unique_values = last_unique_values
+            self.last_unique_values_timestamp = last_unique_values_timestamp
+
+    def check_startup(self) -> List[Tuple[bool, str]]:
+        """
+        Check whether at least one FT sensor message has been received.
+
+        Returns
+        -------
+        startup_status: A list of tuples, where each tuple contains a boolean
+            status of a startup condition and a string describing the condition.
+            All conditions must be True for the startup condition to be considered
+            passed. For example, [(False, "Has received at least one message on
+            topic X")] means that the startup condition has not passed because
+            the node has not received any messages on topic X yet.
+        """
```

Comment:
We discussed changing these messages to "Has not received at least one message on topic X" and "E-stop button has not been clicked at least once" when the conditions aren't satisfied. This will make for easier readability on the app side.